handlers/notify.py

```diff
@@ -1,135 +1,135 @@
-import json
-from os.path import isfile
-from general.values import NOTIFY_DB
+import psycopg2
+import os
+from general.utils import create_sql_array
+import urllib.parse as urlparse
 
 
+url = urlparse.urlparse(os.environ['DATABASE_URL'])
+
 def set_notify_msg(update, context):
-    if not context.args:
-        update.message.reply_text("You need to provide arguments to this command! See /help")
-        return
-    group_id = str(update.effective_chat.id)
-    notify_content = None
-    if not isfile(NOTIFY_DB):
-        f = open(NOTIFY_DB, "w")
-        f.write(json.dumps({}))
-        f.close()
-    with open(NOTIFY_DB, "r") as f:
-        notify_content = json.load(f)
-        new_notify_msg = " ".join(context.args)
-        if notify_content.get(group_id):
-            notify_content[group_id]["message"] = new_notify_msg
-        else:
-            notify_content[group_id] = {
-                "message": new_notify_msg,
-                "tagged": [],
-            }
-
-    with open(NOTIFY_DB, "w") as f:
-        json.dump(notify_content, f, indent=4)
-
-    update.message.reply_text("Done! The new notify message has been set.")
+    with psycopg2.connect(dbname=url.path[1:],user=url.username,password=url.password,host=url.hostname,port=url.port) as conn:
+        if not context.args:
+            update.message.reply_text("You need to provide arguments to this command! See /help")
+            return
+        group_id = str(update.effective_chat.id)
+        with conn.cursor() as cursor:
+            new_notify_msg = " ".join(context.args)
+            cursor.execute('SELECT 1 FROM notifications WHERE id = {}'.format(group_id))
+            group_exists = cursor.fetchone()
+            if group_exists:
+                cursor.execute("UPDATE notifications SET message = '{}' WHERE id = {}".format(new_notify_msg, group_id))
+            else:
+                cursor.execute("INSERT INTO notifications (id, message) VALUES({}, '{}')".format(group_id, new_notify_msg))
```

Comment:
I recommend abstracting out this Database code into its own class. You can have the url variable there, and different methods for fetching, inserting and updating rows. Then from this code you'd call the DB class methods.

You can find more info here:
https://betterprogramming.pub/singleton-in-python-5eaa66618e3d
https://stackoverflow.com/a/60284828